### PR TITLE
Update Quarterdeck Registration API

### PR DESCRIPTION
**Pull request by bbengfort**
### Scope of changes

Update the Registration API with organization information and other data collected from front-end registration form.

Fixes SC-12884

### Type of change

- [ ] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [x] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [ ]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/quarterdeck/db/models/users_test.go [hunk 1]
```diff
@@ -77,13 +91,7 @@ func (m *modelTestSuite) TestUserCreate() {
...
 	require.Equal(int64(3), count, "user count not increased after create")
 
-	// Ensure that the user's role has been created
-	userRole, err := user.UserRole(context.Background())
-	require.NoError(err, "could not fetch user role mapping from database")
-	require.Equal(user.ID, userRole.UserID)
-	require.Equal(int64(2), userRole.RoleID)
-	require.NotEmpty(userRole.Created, "no created timestamp")
-	require.NotEmpty(userRole.Modified, "no modified timestamp")
+	// TODO: Ensure that the user's role has been created
```
Comment (pdeziel) [inline pkg/quarterdeck/db/models/users_test.go:94]: Do we have a story created for this?

File: pkg/quarterdeck/db/models/orgs.go [hunk 2]
```diff
@@ -23,12 +24,17 @@ type Organization struct {
...
 // OrganizationUser is a model representing a many-to-many mapping between users and
-// organizations. This model is primarily used by the User and Organization models and
-// is not intended for direct use generally.
+// organizations and describes the role each user has in their organization. This model
+// is primarily used by the User and Organization models and is not intended for direct
+// use generally.
+//
+// NOTE: a user can only have one role in an organization, so roles must be defined as
+// overlapping sets rather than as disjoint sets where users have multiple roles.
 type OrganizationUser struct {
```
Comment (pdeziel) [inline pkg/quarterdeck/db/models/orgs.go:33]: Is this model currently used?

**Review by pdeziel — APPROVED** [review]
Looks good to me!

File: pkg/quarterdeck/db/models/users_test.go (hunk 1 above)
Comment (bbengfort) [reply pkg/quarterdeck/db/models/users_test.go:94]: Yes, SC-12125

File: pkg/quarterdeck/db/models/orgs.go (hunk 2 above)
Comment (bbengfort) [reply pkg/quarterdeck/db/models/orgs.go:33]: No, not directly, I don't believe except for tests right now.

**Pull request merged**